File: tests/bindings/dotnet/foo.Tests/ClassTest.cs
using System;
using Xunit;
using foo;

namespace foo.Tests
{
    public class ClassTest
    {
        [Fact]
        public void ConstructionDestructionTest()
        {
            Assert.Equal(0u, TestClass.ConstructionCounter());

            var testclass = new TestClass(41);
            Assert.Equal(1u, TestClass.ConstructionCounter());
            Assert.Equal(41u, testclass.GetValue());

            testclass.IncrementValue();
            Assert.Equal(42u, testclass.GetValue());

            testclass.Shutdown();

            Assert.Equal(0u, TestClass.ConstructionCounter());
        }       
    }
}

File: tests/bindings/java/foo-tests/src/test/java/io/stepfunc/foo_test/ClassTest.java
package io.stepfunc.foo_test;

import io.stepfunc.foo.TestClass;
import org.junit.jupiter.api.Test;

import java.util.concurrent.ExecutionException;

import static org.assertj.core.api.Assertions.assertThat;
import static org.joou.Unsigned.uint;

public class ClassTest {
    @Test
    public void ConstructionDestructionTest() {
        assertThat(TestClass.constructionCounter().intValue()).isZero();

        TestClass testclass = new TestClass(uint(41));
        assertThat(TestClass.constructionCounter()).isEqualTo(uint(1));
        assertThat(testclass.getValue()).isEqualTo(uint(41));

        testclass.incrementValue();
        assertThat(testclass.getValue()).isEqualTo(uint(42));

        testclass.delete();

        assertThat(TestClass.constructionCounter().intValue()).isZero();
    }

    @Test
    public void AsyncMethodTest() throws ExecutionException, InterruptedException {
        TestClass testclass = new TestClass(uint(41));
        assertThat(TestClass.constructionCounter()).isEqualTo(uint(1));
        assertThat(testclass.addAsync(uint(1)).toCompletableFuture().get()).isEqualTo(uint(42));

        testclass.incrementValue();
        assertThat(testclass.addAsync(uint(1)).toCompletableFuture().get()).isEqualTo(uint(43));

        testclass.delete();
    }
}
